hexa/workspaces/models.py

```diff
@@ -133,3 +133,16 @@ class WorkspaceMembership(models.Model):
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
     objects = WorkspaceMembershipManager.from_queryset(WorkspaceMembershipQuerySet)()
+
+    def update_if_has_perm(self, *, principal: User, role: WorkspaceMembershipRole):
+        if not principal.has_perm("workspaces_manage_members", self):
+            raise PermissionDenied
+
+        setattr(self, "role", role)
+        return self.save()
+
+    def delete_if_has_perm(self, *, principal: User):
+        if not principal.has_perm("workspaces_manage_members", self):
```

Comment:
From what I read, the permission expects a `Workspace` and not a `WorkspaceMembership` no ?

`workspace.manage_members` is more correct.

Reply:
My bad. You right